src/Simplex/Messaging/Agent/NtfSubSupervisor.hs

```diff
@@ -198,82 +199,154 @@ runNtfWorker c srv Worker {doWork} =
+        splitSubs :: [(NtfSubscription, Either AgentErrorType RcvQueue)] -> ([(ConnId, AgentErrorType)], [CreateSubNtfReq])
+        splitSubs = foldr splitSub ([], [])
+          where
+            splitSub (sub, Right RcvQueue {clientNtfCreds = Just creds}) (errs, ss) = (errs, (sub, creds) : ss)
+            splitSub (sub, Right _) (errs, ss) = ((ntfSubConnId sub, INTERNAL "NSACreate - no notifier queue credentials") : errs, ss)
+            splitSub (sub, Left e) (errs, ss) = ((ntfSubConnId sub, e) : errs, ss)
+        updateSubNSACheck :: DB.Connection -> UTCTime -> (NtfSubscription, NtfSubscriptionId) -> IO ()
+        updateSubNSACheck db checkTs (sub, nSubId) = updateNtfSubscription db sub {ntfSubId = Just nSubId, ntfSubStatus = NASCreated NSNew} (NSANtf NSACheck) checkTs
+    checkSubs :: [NtfSubscription] -> AM' [NtfSubscription]
+    checkSubs ntfSubs =
+      getNtfToken >>= \case
+        Just tkn@NtfToken {ntfServer, ntfTknStatus = NTActive, ntfMode = NMInstant} -> do
+          let (errs1, subsIds) = splitSubs ntfSubs
+          incStatByUserId ntfServer ntfCheckAttempts subsIds
+          rs <- agentNtfCheckSubscriptions c tkn subsIds
+          let (subsIds', errs2, successes) = splitResults rs
+              ntfSubs' = map fst subsIds'
+              errs2' = map (first (ntfSubConnId . fst)) errs2
+              nSubStatuses = map (first fst) successes
+          incStatByUserId ntfServer ntfChecked nSubStatuses
+          ts <- liftIO getCurrentTime
           checkInterval <- asks $ ntfSubCheckInterval . config
           let nextCheckTs = addUTCTime checkInterval ts
-          updateSub (NASCreated toStatus) (NSANtf NSACheck) nextCheckTs
-        updateSub toStatus toAction actionTs' =
-          withStore' c $ \db ->
-            updateNtfSubscription db sub {ntfSubStatus = toStatus} toAction actionTs'
+          (errs3, srvs) <- partitionErrs (ntfSubConnId . fst) nSubStatuses <$> withStoreBatch' c (\db -> map (updateSub db ntfServer ts nextCheckTs) nSubStatuses)
+          mapM_ (getNtfSMPWorker True c) $ S.fromList (catMaybes srvs)
+          workerErrors c $ errs1 <> errs2' <> errs3
+          pure ntfSubs'
+        _ -> do
+          let errs = map (\sub -> (ntfSubConnId sub, INTERNAL "NSACheck - no active token")) ntfSubs
+          workerErrors c errs
+          pure []
+      where
+        splitSubs :: [NtfSubscription] -> ([(ConnId, AgentErrorType)], [CheckSubNtfReq])
+        splitSubs = foldr splitSub ([], [])
+          where
+            splitSub sub@NtfSubscription {ntfSubId = Just nSubId} (errs, ss) = (errs, (sub, nSubId) : ss)
+            splitSub sub (errs, ss) = ((ntfSubConnId sub, INTERNAL "NSACheck - no subscription ID") : errs, ss)
+        updateSub :: DB.Connection -> NtfServer -> UTCTime -> UTCTime -> (NtfSubscription, NtfSubStatus) -> IO (Maybe SMPServer)
+        updateSub db ntfServer ts nextCheckTs (sub@NtfSubscription {smpServer}, status) = case status of
+          NSAuth -> do
```

Comment:
perhaps we should do same on NSDeleted - it's a new status for receiving DELD (currently we don't)

Reply:
or even use ntfShouldSubscribe